third_party/ObjexxFCL/src/ObjexxFCL/vectorize.hh

```diff
@@ -23,16 +23,33 @@
 #define ASSUME(b)
 #define ASSUME_ALIGNED(p,b) p=(decltype(p))__builtin_assume_aligned(p,b)
 #define RESTRICT __restrict__
-#elif defined(_WIN32) && ( defined(_MSC_VER) || defined(__INTEL_COMPILER) )
+
+#elif defined(_WIN64) && ( defined(_MSC_VER) || defined(__INTEL_COMPILER) )
 #define ASSUME(b) __assume(b)
 #define ASSUME_ALIGNED(p,b) __assume_aligned(p,b)
 #define RESTRICT __restrict
+
+// Disable alignment hints for 32-bit windows builds
+#elif defined(_WIN32) && ( defined(_MSC_VER) || defined(__INTEL_COMPILER) )
+#define ASSUME(b) __assume(b)
+#define ASSUME_ALIGNED(p,b) 
+#define RESTRICT __restrict
+
 #else
 #define ASSUME(b)
 #define ASSUME_ALIGNED(p,b)
 #define RESTRICT
 #endif
 
+// Alignment for Compiler Options
+#if defined (__AVX2__)
```

Comment:
@amirroth sure, but I really meant things not used by ObjexxFCL: VEC_ALIGN and disabling the ASSUME and RESTRICT macros in 32-bit Windows aren't things that should be part of the library but we can make that happen in the EnergyPlus vectorize.hh. Anyway, I can adjust this before the final merge.

I think we also need to disable the explicit vectorization on 32-bit Linux where we also don't have a guarantee of 16 byte native heap alignment.